base/tuple.jl

```diff
@@ -105,7 +105,8 @@ julia> ntuple(i -> 2*i, 4)
 ```
 """
 function ntuple(f::F, n::Integer) where F
-    t = n <= 0  ? () :
+    (n >= 0) || throw(DomainError())
```

Comment:
I would be surprised if LLVM did not just hoist the branch to the end, but this change can't hurt.